### PR TITLE
Pulling out the inputs from the Spec test fixture

### PR DESCRIPTION
**Pull request by JeffAshton**

### CODE REVIEW

File: tests/D2L.CodeStyle.Analyzers.Test/SpecRunner.cs [hunk 1]
```diff
@@ -17,8 +8,10 @@
 using NUnit.Framework;
 
 namespace D2L.CodeStyle.Analyzers {
-	[TestFixtureSource( nameof( m_specNames ) )]
+
+	[TestFixtureSource( typeof( SpecTestsProvider ), nameof( SpecTestsProvider.GetAll ) )]
```
Comment (JeffAshton) [inline tests/D2L.CodeStyle.Analyzers.Test/SpecRunner.cs:12]: Pulling all the inputs that were stored in statics out of fixture into test assembly specific `SpecTestsProvider`:

https://github.com/Brightspace/D2L.CodeStyle/pull/847/files#diff-200403da8446b3133b2a790dac3a198033138dae29d25a1151e707564f0a4435R10

File: tests/D2L.CodeStyle.Analyzers.Test/SpecRunner.cs [hunk 2]
```diff
@@ -39,52 +32,10 @@ private sealed class DiagnosticComparer : IEqualityComparer<Diagnostic> {
...
-				} )
-				.ToImmutableDictionary( nv => nv.Name, nv => nv.Value );
-
-			foreach( var diag in m_possibleDiagnostics ) {
-				Assert.NotNull( diag.Value, $"Common.Diagnostics.{diag.Key} must be of type DiagnosticDescriptor and not null" );
-			}
-		}
-
 		/// <summary>
 		/// Each spec causes a single compilation to happen. This constructor
```
Comment (JeffAshton) [inline tests/D2L.CodeStyle.Analyzers.Test/SpecRunner.cs:40]: I clean up this comment in a later commit

File: tests/D2L.CodeStyle.Analyzers.Test/SpecTestsProvider.cs [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+namespace D2L.CodeStyle.Analyzers {
+
+	internal static class SpecTestsProvider {
+
+		public static IEnumerable<SpecTest> GetAll() {
+
+			ImmutableArray<AdditionalText> additionalFiles = GetAdditionalFiles();
+			ImmutableDictionary<string, DiagnosticDescriptor> diagnosticDescriptors = GetDiagnosticDescriptors();
+			ImmutableArray<MetadataReference> metdataReferences = GetMetadataReferences().ToImmutableArray();
```
Comment (omsmith) [inline tests/D2L.CodeStyle.Analyzers.Test/SpecTestsProvider.cs:16]:
```suggestion
			ImmutableArray<MetadataReference> metdataReferences = GetMetadataReferences();

```

**Review by omsmith — APPROVED** [review]

File: tests/D2L.CodeStyle.Analyzers.Test/SpecTestsProvider.cs [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+				m_text = text;
+			}
+
+			public override string Path { get; }
+			public override SourceText GetText( CancellationToken cancellationToken = default ) => SourceText.From( m_text, Encoding.UTF8 );
+		}
+
+		private static string ReadEmbeddedResourceAsString( Assembly assembly, string name ) {
+
+			using( Stream stream = assembly.GetManifestResourceStream( name ) )
```
Comment (j3parker) [inline tests/D2L.CodeStyle.Analyzers.Test/SpecTestsProvider.cs:130]: nit: there are statements now (`using var foo = ...`)

**Review by j3parker — APPROVED** [review]

File: tests/D2L.CodeStyle.Analyzers.Test/SpecTestsProvider.cs (hunk 4 above)
Comment (JeffAshton) [reply tests/D2L.CodeStyle.Analyzers.Test/SpecTestsProvider.cs:130]: Good catch.  It was from here:

image

Comment (JeffAshton) [reply tests/D2L.CodeStyle.Analyzers.Test/SpecTestsProvider.cs:130]: I'll fix in a follow up PR

**Pull request merged**